src/map/lua/luautils.cpp

```diff
@@ -3789,6 +3792,39 @@ namespace luautils
         return result.get_type(0) == sol::type::number ? result.get<int32>(0) : 0;
     }
 
+    int32 OnPetAbility(CBaseEntity* PTarget, CPetEntity* PPet, CPetSkill* PPetSkill, CBaseEntity* PMobMaster, action_t* action)
+    {
+        TracyZoneScoped;
+
+        std::string filename = fmt::format("./scripts/globals/abilities/pets/{}.lua", PPetSkill->getName());
```

Comment:
Do these hot reload correctly?

Reply:
They do, it's sourced the same way as current pet abilities used by SMN are. Can confirm that I can edit Welt in real time and it reloads.